CHANGES.md

```diff
@@ -2,6 +2,434 @@
+-------------------------------------------
+* added a new --denormalize flag that cleans up generated code before codegen
+* stopped throwing --vectorize when --fast is thrown
+
+Portability Improvements
+------------------------
+* improved the portability of the code base to gcc 6.x
+* improved portability to FreeBSD, Solaris, illumos, OmniOS, Ubuntu
+* added initial support for compiling single-locale programs for 64-bit ARM
+
+Cray-specific Changes
+---------------------
+* removed an obsolete warning about having hugepage modules loaded on Crays
+
+Tool Changes
+------------
+* 'chplvis' improvements:
+  - added a "settings" window
+  - fixed undetected 'chplvis' regressions in 1.13
+  - made significant improvements to the user interface
+  - added new views of execution-time data
+  (see TODO)
+* 'chpldoc' improvements: improved formatting and handling of many situations
+
+Syntax Highlighting
+-------------------
+* added instructions for using vim highlighting for Vundle and vim-plug users
+* fixed portability of Chapel's emacs mode for emacs 24.x
+* added support for nested block comments to emacs and 'highlight' highlighting
+* fixed the highlighting of binary literals in Pygments
+* updated our support for Andre Simon's highlighter to support version 3.x
+
+Error Message Improvements
+--------------------------
+* added a clarifying statement for 'const' errors due to task/forall intents
+* made string OOB messages print the offending index
+* squashed C warnings considered to be noise in the back-end compiler
+* added an error message for declaring constructors/initializers without parens
+* added an error for user fields named 'outer'
+* fixed some spelling errors in error messages
+* removed warnings for assigning serial iterators/iterables to arrays
+* removed a misleading warning about records and forall intents
+
+Runtime Error Checks
+--------------------
+* added bounds checking to string index/slice operations
+
+Bug Fixes
+---------
+* fixed a bug in which having CONFIG_SITE set could break third-party builds
+* made 'compare_exchange' conform to memory order requirements on failure
+* fixed incorrect 'fetchAdd' and 'fetchSub' implementations on atomics
+* fixed a bug in which 64-bit atomic loads on 32-bit systems were not atomic
+* fixed a longstanding but infrequent race condition in privatization
+* fixed a bug in handling 'ref' return intents
+* fixed a bug for recursive functions that return records
+* fixed a bug in accessing overridden methods
+* fixed a bug in which reduce intents could not be applied to arrays
+* fixed a bug in iterating over strided, unsigned domains and arrays
+* fixed a bug in BlockCyclic array accesses for 3D+ arrays
+* fixed a bug for membership queries in sparse CSR layouts
+* fixed a bug when storing max(int) in an associative domain of ints
+* fixed a bug in which arrays within records were incorrectly being localized
+* fixed a bug in which (+/-1)**k was incorrect for negative values of k
```

Comment:
I think the '**' here may be causing the rest of the document to be rendered as bold text, if that matters.